### PR TITLE
Ipyleaflet providers and fix for compress providers

### PR DESCRIPTION
**Pull request by sackh**
I have added some providers which were missing here and present in [ipyleaflet](https://github.com/jupyter-widgets/ipyleaflet/blob/master/ipyleaflet/basemaps.py)
Also while adding provider found an issue in `provider_sources/_compress_providers.py`
If Group is already present and if we add new basemaps to the same group then it is overriding the dictionary due to line https://github.com/geopandas/xyzservices/blob/1f6ac89f4aaf7ab04ab40f5c659c32f100b8c3c6/provider_sources/_compress_providers.py#L52

example:
`OpenStreetMap` group is already present, If I add a new entry in this group then as per existing code it will replace the value and will keep only the new value, it is not updating the dictionary.
Please correct me if I am wrong.
I have also created PR # https://github.com/jupyter-widgets/ipyleaflet/pull/857 in ipyleaflet to support xyzservices.

### CODE REVIEW

File: provider_sources/_compress_providers.py [hunk 1]
```diff
@@ -49,7 +49,18 @@
         )
 
 # combine both
-leaflet.update(xyz)
+
+for key, val in xyz.items():
+    if key in leaflet:
+        if any(
+            isinstance(i, dict) for i in leaflet[key].values()
+        ):  # for related group of bunch
```
Comment (martinfleis) [inline provider_sources/_compress_providers.py:57]:
```suggestion
        if isinstance(val, Bunch): 
```

Is this what you're trying to get here? I think we can easily import `xyzservices.Bunch` above here and check directly. We'll just need to make sure that we have xyzservices install in our GHA here https://github.com/geopandas/xyzservices/blob/main/.github/workflows/update_providers.yaml (I can take care of that later).

File: xyzservices/data/providers.json [hunk 2]
```diff
@@ -2362,6 +2369,34 @@
             "variant": "MODIS_Terra_Chlorophyll_A",
             "opacity": 0.75,
             "name": "NASAGIBS.ModisTerraChlorophyll"
+        },
+        "ModisTerraBands721CR": {
+            "url": "https://gibs.earthdata.nasa.gov/wmts/epsg3857/best/MODIS_Terra_CorrectedReflectance_Bands721/default/{time}/GoogleMapsCompatible_Level9/{z}/{y}/{x}.jpg",
+            "max_zoom": 9,
+            "attribution": "Imagery provided by services from the Global Imagery Browse Services (GIBS), operated by the NASA/GSFC/Earth Science Data and Information System (<a href=\"https://earthdata.nasa.gov\">ESDIS</a>) with funding provided by NASA/HQ.",
+            "html_attribution": "Imagery provided by services from the Global Imagery Browse Services (GIBS), operated by the NASA/GSFC/Earth Science Data and Information System (<a href=\"https://earthdata.nasa.gov\">ESDIS</a>) with funding provided by NASA/HQ.",
+            "name": "NASAGIBS.ModisTerraBands721CR"
```
Comment (martinfleis) [inline xyzservices/data/providers.json:2378]:
```suggestion
            "name": "NASAGIBS.ModisTerraBands721CR",
            "time": ""
```
If there's a `time` placeholder we need to define an attribute for that.

File: xyzservices/data/providers.json [hunk 3]
```diff
@@ -2362,6 +2369,34 @@
...
+        },
+        "ModisAquaTrueColorCR": {
+            "url": "https://gibs.earthdata.nasa.gov/wmts/epsg3857/best/MODIS_Aqua_CorrectedReflectance_TrueColor/default/{time}/GoogleMapsCompatible_Level9/{z}/{y}/{x}.jpg",
+            "max_zoom": 9,
+            "attribution": "Imagery provided by services from the Global Imagery Browse Services (GIBS), operated by the NASA/GSFC/Earth Science Data and Information System (<a href=\"https://earthdata.nasa.gov\">ESDIS</a>) with funding provided by NASA/HQ.",
+            "html_attribution": "Imagery provided by services from the Global Imagery Browse Services (GIBS), operated by the NASA/GSFC/Earth Science Data and Information System (<a href=\"https://earthdata.nasa.gov\">ESDIS</a>) with funding provided by NASA/HQ.",
+            "name": "NASAGIBS.ModisAquaTrueColorCR"
```
Comment (martinfleis) [inline xyzservices/data/providers.json:2385]:
```suggestion
            "name": "NASAGIBS.ModisAquaTrueColorCR",
            "time": ""
```

File: xyzservices/data/providers.json [hunk 4]
```diff
@@ -2362,6 +2369,34 @@
...
+        },
+        "ModisAquaBands721CR": {
+            "url": "https://gibs.earthdata.nasa.gov/wmts/epsg3857/best/MODIS_Aqua_CorrectedReflectance_Bands721/default/{time}/GoogleMapsCompatible_Level9/{z}/{y}/{x}.jpg",
+            "max_zoom": 9,
+            "attribution": "Imagery provided by services from the Global Imagery Browse Services (GIBS), operated by the NASA/GSFC/Earth Science Data and Information System (<a href=\"https://earthdata.nasa.gov\">ESDIS</a>) with funding provided by NASA/HQ.",
+            "html_attribution": "Imagery provided by services from the Global Imagery Browse Services (GIBS), operated by the NASA/GSFC/Earth Science Data and Information System (<a href=\"https://earthdata.nasa.gov\">ESDIS</a>) with funding provided by NASA/HQ.",
+            "name": "NASAGIBS.ModisAquaTrueColorCR"
```
Comment (martinfleis) [inline xyzservices/data/providers.json:2392]:
```suggestion
            "name": "NASAGIBS.ModisAquaTrueColorCR",
            "time": ""
```

File: xyzservices/data/providers.json [hunk 5]
```diff
@@ -2362,6 +2369,34 @@
...
+        },
+        "ViirsTrueColorCR": {
+            "url": "https://gibs.earthdata.nasa.gov/wmts/epsg3857/best/VIIRS_SNPP_CorrectedReflectance_TrueColor/default/{time}/GoogleMapsCompatible_Level9/{z}/{y}/{x}.jpg",
+            "max_zoom": 9,
+            "attribution": "Imagery provided by services from the Global Imagery Browse Services (GIBS), operated by the NASA/GSFC/Earth Science Data and Information System (<a href=\"https://earthdata.nasa.gov\">ESDIS</a>) with funding provided by NASA/HQ.",
+            "html_attribution": "Imagery provided by services from the Global Imagery Browse Services (GIBS), operated by the NASA/GSFC/Earth Science Data and Information System (<a href=\"https://earthdata.nasa.gov\">ESDIS</a>) with funding provided by NASA/HQ.",
+            "name": "NASAGIBS.ViirsTrueColorCR"
```
Comment (martinfleis) [inline xyzservices/data/providers.json:2399]:
```suggestion
            "name": "NASAGIBS.ViirsTrueColorCR",
            "time": ""
```

**Review by martinfleis — COMMENTED** [review]
Thanks a lot! 

A few comments:

- additional providers need to be defined in [`provider_sources/xyzservices-providers.json`](https://github.com/geopandas/xyzservices/blob/main/provider_sources/xyzservices-providers.json). `xyzservices/data/providers.json` is autogenerated.
- are you sure that Strava tiles actually work? They fail on our CI and seem to be down according to https://ipyleaflet.readthedocs.io/en/latest/api_reference/basemaps.html as well. If you think it makes sense to keep them, can you give them `"status": "broken"` attribute?
- ModisTerraBands721CR, ModisAquaTrueColorCR and ViirsTrueColorCR need `time` attribute as there's a `{time}` placeholder
- and a few notes in the code

File: xyzservices/data/providers.json (hunk 2 above)
Comment (sackh) [reply xyzservices/data/providers.json]: done

File: provider_sources/_compress_providers.py (hunk 1 above)
Comment (sackh) [reply provider_sources/_compress_providers.py]: Yes, done

File: xyzservices/data/providers.json (hunk 3 above)
Comment (sackh) [reply xyzservices/data/providers.json]: done

File: xyzservices/data/providers.json (hunk 4 above)
Comment (sackh) [reply xyzservices/data/providers.json]: done

File: xyzservices/data/providers.json (hunk 5 above)
Comment (sackh) [reply xyzservices/data/providers.json]: done

**Comment by martinfleis** [comment]
@sackh Thanks! I have moved the added metadata to `provider_sources/xyzservices-providers.json` so they are not lost after `make compress` and reverted the suggested change in the compress script because it was obviously a wrong suggestion from my side. Also added a changelog.

**Pull request merged**